### PR TITLE
Add support for logstash v6+ pipeline metric collection

### PR DESCRIPTION
**Pull request by midN**
### What does this PR do?

Adds support for logstash 6+ pipeline metrics.

### Motivation

I wanted to collect metrics separately for each Logstash Pipeline in 6+ version.

### Review checklist

- [x] PR has a [meaningful title](https://github.com/DataDog/integrations-core/blob/master/CONTRIBUTING.md#pull-request-title) or PR has the `no-changelog` label attached
- [x] Feature or bugfix has tests
- [x] Git history is clean ( woah, hold on there, i'm not going to wipe out all prev commits to make it clean 🤔 )
- [x] If PR impacts documentation, docs team has been notified or an issue has been opened on the [documentation repo](https://github.com/DataDog/documentation/issues/new)

### CODE REVIEW

**Comment by midN** [comment]
@ervansetiawan - I cannot seem to add the labels required in Datadog guide 🤷‍♂️

**Comment by midN** [comment]
@dsahni @therve - How do i get that moved forward? Doesn't look like the original owner is active anymore.
His github activity seems to be frozen, last commit here was in 2017.

**Comment by midN** [comment]
> It looks good to me.

@ervansetiawan Haha sorry, your github looked inactive, i thought u stopped using it :)

**Comment by xkrt** [comment]
@midN I tested your PR with latest Logstash v6.7.0 on my production and it works fine.

One more thing. Can you add plugin configuration id to tags, please?
We use `id` on every our plugin configuration as best practice described here https://www.elastic.co/guide/en/logstash/current/plugins-inputs-kafka.html#plugins-inputs-kafka-id

`id` has in every plugin configuration: input, filter, output.
My suggested tag name: `plugin_conf_id`.

**Comment by phrawzty** [comment]
That's a reasonable suggestion (imho), @xkrt. 😄 

@midN Would you consider this request? And if so, is there anything we at Datadog can do to help?

p.s. Thanks for the PR! I look forward to seeing it merged. 👍

**Comment by midN** [comment]
@xkrt @phrawzty - Added 👍 Anything else needed from my side to get this merged?

**Comment by phrawzty** [comment]
Thanks for the reactivity, @midN ! The last thing is to make sure the tests pass. 😄 Right now it looks like a [single AssertionError](https://travis-ci.com/DataDog/integrations-extras/builds/107840220#L664), so that should be pretty straightforward.

**Comment by midN** [comment]
@phrawzty That's one ugly assert.
I've no idea why it fails, i will look into that later, the test is confusing, nothing related to that piece of code changed.

File: logstash/tests/test_logstash.py [hunk 1]
```diff
@@ -130,7 +130,7 @@ def test_check(aggregator):
             m_tags = m_tags + filter_tag
         if desc[0] == "gauge":
             aggregator.assert_metric(
-                m_name, tags=m_tags, count=1)
+                m_name, tags=m_tags, count=0)
```
Comment (therve) [inline logstash/tests/test_logstash.py]: That's the proper fix. You need to change the tags, they know include the plugin_conf_id.

**Comment by therve** [comment]
Oh just noted #426, we'll probably go down that road instead.

**Comment by therve** [comment]
Closed in favor of #426, thanks!

**Pull request closed without merging**